src/libsyncengine/update_detection/file_system_observer/computefsoperationworker.cpp

```diff
@@ -571,7 +571,8 @@ void ComputeFSOperationWorker::logOperationGeneration(const ReplicaSide side, co
 }
 
 ExitCode ComputeFSOperationWorker::checkFileIntegrity(const DbNode &dbNode) {
-    if (dbNode.type() == NodeType::File && dbNode.nodeIdLocal().has_value() && dbNode.nodeIdRemote().has_value() &&
+    if (CommonUtility::isFileSizeMismatchDetectionEnabled() && dbNode.type() == NodeType::File && dbNode.nodeIdLocal().has_value() &&
+        dbNode.nodeIdRemote().has_value() &&
         dbNode.lastModifiedLocal().has_value()) {
         if (_fileSizeMismatchMap.find(dbNode.nodeIdLocal().value()) != _fileSizeMismatchMap.end()) {
```

Comment:
An opportunity to use `contains` instead of `find`.

Reply:
Done here: [f555608](https://github.com/Infomaniak/desktop-kDrive/pull/269/commits/f5556083c32621bc2c1b2944a7197c214195e03d)